Lib/_pyrepl/base_eventqueue.py

```diff
@@ -69,18 +69,14 @@ def insert(self, event: Event) -> None:
         trace('added event {event}', event=event)
         self.events.append(event)
 
-    def push(self, char: int | bytes | str) -> None:
+    def push(self, char: int | bytes) -> None:
         """
         Processes a character by updating the buffer and handling special key mappings.
         """
+        assert isinstance(char, (int, bytes))
         ord_char = char if isinstance(char, int) else ord(char)
-        if ord_char > 255:
-            assert isinstance(char, str)
-            char = bytes(char.encode(self.encoding, "replace"))
-            self.buf.extend(char)
-        else:
-            char = bytes(bytearray((ord_char,)))
-            self.buf.append(ord_char)
+        char = bytes((ord_char,))
```

Comment:
More explicit perhaps?
```suggestion
        char = ord_char.to_bytes()
```

Reply:
Yeah, that's what I've used in my draft https://github.com/python/cpython/pull/133030 in a few places, too.

Reply:
@tomasr8 Thanks! But I hope we can sometimes use solution from  #133030 :)